### PR TITLE
Fix case that was causing us to not offer completion after a case-pattern-label in a switch.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/15348

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@gafter I fixed this myself as i was uncertain if you'd be able to get to this.  Thanks!

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide .  This first commit contains the fix.  The second commit is cleanup.

File: src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTokenExtensions.cs [hunk 1]
```diff
@@ -99,91 +99,74 @@ public static bool IsBeginningOfStatementContext(this SyntaxToken token)
             // lock (expr)
             //   |
 
-            // for ( ; ; Foo(), |
```
Comment (sharwell) [inline src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTokenExtensions.cs]: ❔ Did you mean to remove this comment?

**Comment by CyrusNajmabadi** [comment]
Review may be easier adding ?w=1 to the end.

File: src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTokenExtensions.cs (hunk 1 above)
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTokenExtensions.cs]: Yes.  it does not apply.  That's not a statement location (i.e. you can't type 'return' there).  This is also why there's no case below that handles hte .Comma case.

**Comment by CyrusNajmabadi** [comment]
**Customer scenario**

What does the customer do to get into this situation, and why do we think this
is common enough to address for this release.  (Granted, sometimes this will be
obvious "Open project, VS crashes" but in general, I need to understand how
common a scenario is)

**Bugs this fixes:**

(either VSO or GitHub links)

**Workarounds, if any**

Also, why we think they are insufficient for RC vs. RC2, RC3, or RTW

**Risk**

This is generally a measure our how central the affected code is to adjacent
scenarios and thus how likely your fix is to destabilize a broader area of code

**Performance impact**

(with a brief justification for that assessment (e.g. "Low perf impact because no extra allocations/no complexity changes" vs. "Low")

**Is this a regression from a previous update?**

**Root cause analysis:**

How did we miss it?  What tests are we adding to guard against it in the future?

**How was the bug found?**

(E.g. customer reported it vs. ad hoc testing)

**Pull request closed without merging**